README.md

```diff
@@ -12,20 +12,45 @@ Offers alternative types of file parameter that are compatible with Pipeline and
 
 See [JENKINS-27413](https://issues.jenkins-ci.org/browse/JENKINS-27413) and [JENKINS-29289](https://issues.jenkins-ci.org/browse/JENKINS-29289) for background.
 
+## Usage in declarative pipeline
+
+You can now declare file parameters as is in declarative pipeline:
+
+```groovy
+pipeline {
+  agent any
+  parameters {
+    base64File 'FILE'
+    stashedFile 'FILE-STASH'
+  }
+  stages {
+    stage('Example') {
+      steps {
+        withFileParameter('FILE') {
+          sh 'cat $FILE'
+        }
+        unstash 'FILE-STASH'
+        echo(/loaded '${readFile('FILE-STASH')}'/)
+      }
+    }
+  }
+}
+```
+
 ## Implementation status
 
 - [X] Base64 file parameter (simple and suitable for small files)
   - [X] implementation
-  - [ ] inline help text
+  - [X] inline help text
   - [X] `withFileParameter` compatibility
   - [X] manual test
   - [X] automated test
 - [ ] stashed file parameter (suitable for larger files used in Pipeline)
   - [X] implementation
-  - [ ] inline help text
-  - [ ] `withFileParameter` compatibility
+  - [X] inline help text
+  - [X] `withFileParameter` compatibility (manual test)
```

Comment:
I do not think it could possibly work currently: https://github.com/jenkinsci/file-parameters-plugin/blob/b2e08aa74bb2d655c92a989c2fa191dd6fc16cf4/src/main/java/io/jenkins/plugins/file_parameters/StashedFileParameterValue.java#L83

Reply:
well not sure using such usage is very useful :) but true we can be error prone...

```
pipeline {
    agent any
    parameters {
        stashedFile(name: 'FILE')
    }
    stages {
        stage('Example') {
            steps {
                withFileParameter(name:'FILE') {
                    unstash "FILE"
                    sh 'cat $FILE'
                }
            }
        }
    }
}

Reply:
If you are using `withFileParameter` then you are not supposed to need `unstash` (and vice-versa).

Reply:
```suggestion
  - [ ] `withFileParameter` compatibility
```